nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -4589,6 +4612,13 @@ mod tests {
             self.counts.values().all(|&count| count == 0)
         }
 
+        /// Add counts from another BlueprintTableCounts to this one.
```

Comment:
What do you think about having `BlueprintTableCounts` contain an integer counter of the number of blueprints we think should be here?  `new()` would set it to 1, and `add()` would add the count from the `other`.

Then in `ensure_blueprint_fully_populated()`, for the non-exception tables, their counts should match the count in the struct.

Reply:
That totally make sense, thank you! Will add it shortly.